src/pkg/sources/new.go

```diff
@@ -17,8 +18,17 @@ import (
 )
 
 // New creates a new package source based on pkgLocation
-func New(pkgLocation string, pkg types.Package, opts zarfTypes.ZarfPackageOptions, sha string, nsOverrides NamespaceOverrideMap) (zarfSources.PackageSource, error) {
+func New(bundleCfg types.BundleConfig, pkg types.Package, opts zarfTypes.ZarfPackageOptions, sha string, nsOverrides NamespaceOverrideMap) (zarfSources.PackageSource, error) {
 	var source zarfSources.PackageSource
+	var pkgLocation string
+	if bundleCfg.DeployOpts.Source != "" {
```

Comment:
Is it necessary to check that the `.Source != ""` or is it possible to nil check `bundleCfg.DeployOpts`? Like, if your calling remove, is `DeployOpts` nil?

Reply:
Good question, but just double checked and the unused "Opts" are not nil, so on Deploy, `RemoveOpts` has `Source` set to "" and `Packages` set to an empty list.